timelock-impl/src/test/java/com/palantir/atlasdb/timelock/adjudicate/FeedbackSinkTest.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * (c) Copyright 2020 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.timelock.adjudicate;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+import java.util.List;
+import java.util.UUID;
+import java.util.concurrent.TimeUnit;
+
+import org.junit.Test;
+
+import com.github.benmanes.caffeine.cache.Caffeine;
+import com.palantir.timelock.feedback.ConjureTimeLockClientFeedback;
+
+public class FeedbackSinkTest {
+    private static final FakeTicker FAKE_TICKER = new FakeTicker();
+    private static final ConjureTimeLockClientFeedback TEST_REPORT = getTestReport();
+
+    @Test
+    public void feedbackIsRegistered() {
+        List<ConjureTimeLockClientFeedback> trackedFeedbackReports =
+                createSinkAndAddTestReport().getTrackedFeedbackReports();
+        assertThat(trackedFeedbackReports.size()).isEqualTo(1);
+        assertThat(trackedFeedbackReports).containsExactly(TEST_REPORT);
+    }
+
+    @Test
+    public void feedbackRetrievedFromSinkIsImmutable() {
+        List<ConjureTimeLockClientFeedback> trackedFeedbackReports =
+                createSinkAndAddTestReport().getTrackedFeedbackReports();
+        assertThatThrownBy(() -> trackedFeedbackReports.add(TEST_REPORT))
+                .isInstanceOf(UnsupportedOperationException.class);
+    }
+
+    @Test
+    public void feedbackReportIsEvictedAfterExpiry() {
+        TimeLockClientFeedbackSink sink = createSinkAndAddTestReport();
+        List<ConjureTimeLockClientFeedback> trackedFeedbackReports =
+                sink.getTrackedFeedbackReports();
+        assertThat(trackedFeedbackReports.size()).isEqualTo(1);
+        assertThat(trackedFeedbackReports).containsExactly(TEST_REPORT);
+
+        FAKE_TICKER.advance(Constants.HEALTH_FEEDBACK_REPORT_EXPIRATION_MINUTES, TimeUnit.MINUTES);
+        assertThat(sink.getTrackedFeedbackReports().size()).isEqualTo(0);
+    }
+
```

Comment:
I think there should be a test for multiple feedback reports as well - add 100 and verify they're still there.